src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -246,8 +193,18 @@ var TestRenderer = ReactFiberReconciler({
   useSyncScheduling: true,
 
   getPublicInstance(ref) {
-    const createNodeMock = ref.rootContainerInstance.createNodeMock;
-    return createNodeMock(ref);
+    if (typeof ref.text === 'string') {
+      return ref.text;
+    } else if (
+      typeof ref.rootContainerInstance === 'object' &&
+      typeof ref.rootContainerInstance.createNodeMock === 'function'
+    ) {
+      const createNodeMock = ref.rootContainerInstance.createNodeMock;
+      return createNodeMock(ref);
+    }
+
+    // this should not be possible
+    throw new Error('Attempted to getPublicInstance on an invalid ref.');
```

Comment:
Like [this](https://github.com/facebook/react/blob/1c43f9471773a5206279236c5e8b5b37d3e985cb/src/renderers/shared/fiber/ReactChildFiber.js#L397-L422)?